internal/ingest/adchainprocessor/adchainprocessor.go

```diff
@@ -0,0 +1,404 @@
+package adchainprocessor
+
+// TODO
+// 4. Remove the other dead code.
+// 6. More tests
+// 8. Differntiate between publisher/provider
+
+// TODO namings
+// s/entryChunk/entryChunk
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/filecoin-project/storetheindex/internal/metrics"
+	"github.com/ipfs/go-cid"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/codec/dagjson"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	selectorparse "github.com/ipld/go-ipld-prime/traversal/selector/parse"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+)
+
+var log = logging.Logger("indexer/ingest/adchainprocessor")
+
+const defaultUpdateProcessorUpdateChanBufferSize = 3
```

Comment:
Why 3?

>I believe there are only two values for a channel's buffer, 0 and 1.
Setting a value higher than 1 is really saying "well, I sort of want to
buffer as much as I can, I hope this number is enough", and that is not a
stable way to design systems.
  -- Dave Cheney

Reply:
Agreed. I think I copied this from some similar logic, but I agree it should just be 1.